lib/Runtime/Library/BoundFunction.cpp

```diff
@@ -171,9 +172,20 @@ namespace Js
             }
 
             // Copy the bound args
-            for (uint i=0; i<boundFunction->count; i++)
+            if (!isCrossSiteObject)
             {
-                newValues[index++] = boundFunction->boundArgs[i];
+                for (uint i = 0; i < boundFunction->count; i++)
```

Comment:
We'll do a quick chceck at the beginning of MarshalVar. Want to avoid the check in general case.